### PR TITLE
feat(webjars): use webjars to obtain JS dependencies 

### PR DESCRIPTION
**Pull request by maximilianhuber**
this PR introduces webjars, which are used to pull the JS dependencies by maven.

The URIs were modified via the following script:
```bash
while read repl; do
    find ./frontend/sw360-portlet/src/main/webapp/html/ -type f -print0 | \
        xargs -0 sed -i 's%'"$repl"'%g'
    sed -i 's%'"$repl"'%g' frontend/sw360-portlet/src/main/webapp/WEB-INF/liferay-portlet.xml
done <<EOM
/js/external/jquery-1.11.1.min.js%/webjars/jquery/1.12.4/jquery.min.js
/js/external/jquery-ui.min.js%/webjars/jquery-ui/1.12.1/jquery-ui.min.js
/js/external/jquery.dataTables-1.9.4.js%/webjars/datatables/1.10.7/js/jquery.dataTables.min.js
/js/external/jquery.dataTables.js%/webjars/datatables/1.10.7/js/jquery.dataTables.min.js
/js/external/jquery-confirm.min.js%/webjars/github-com-craftpip-jquery-confirm/3.0.1/jquery-confirm.min.js
/js/external/jquery.treetable.js%/webjars/jquery-treetable/3.2.0/jquery.treetable.js
/js/external/resumable.js%/webjars/resumable.js/1.0.2/resumable.js
/js/external/jquery.validate.min.js%/webjars/jquery-validation/1.15.1/jquery.validate.min.js
/js/external/additional-methods.min.js%/webjars/jquery-validation/1.15.1/additional-methods.min.js
/css/external/jquery.treetable.css%/webjars/jquery-treetable/3.2.0/css/jquery.treetable.css
/css/external/jquery.treetable.theme.default.css%/webjars/jquery-treetable/3.2.0/css/jquery.treetable.theme.default.css
/css/external/jquery-ui.css%/webjars/jquery-ui/1.12.1/jquery-ui.css
/css/external/jquery-confirm.min.css%/webjars/github-com-craftpip-jquery-confirm/3.0.1/jquery-confirm.min.css
EOM
```

### CODE REVIEW

**Comment by maximilianhuber** [comment]
This solves https://github.com/sw360/sw360portal/issues/284

File: frontend/sw360-portlet/src/main/webapp/js/main.js [hunk 1]
```diff
@@ -240,9 +240,11 @@ function deleteConfirmed(confirmMessage, confirmCallback) {
         content: confirmMessage,
         confirmButtonClass: 'btn-info',
         cancelButtonClass: 'btn-danger',
-        confirm: function() {
-            confirmCallback();
-        }
+        buttons: {
```
Comment (alex-evo) [inline frontend/sw360-portlet/src/main/webapp/js/main.js]: This creates a confirm dialog with no option to cancel. I see you have bumped the version of jquery-confirm, but didn't sufficiently adapt to its new API. Please make sure to restore the functionality.

**Comment by alex-evo** [comment]
I also wonder whether there are more such compatibility problems with other libraries, that I have missed. I think we should brace ourselves for some JS-related UI bugs that might have been introduced by bumping JS libraries' versions. I believe it's worth it nevertheless.

File: frontend/sw360-portlet/src/main/webapp/js/main.js (hunk 1 above)
Comment (maximilianhuber) [inline frontend/sw360-portlet/src/main/webapp/js/main.js]: I have followed the instructions at https://github.com/craftpip/jquery-confirm/wiki/Migrate-from-v2-to-v3. Also in our old code there was no cancel mentioned in the code. I will check this.

Can you give me a hint where this dialog appears?

Comment (alex-evo) [inline frontend/sw360-portlet/src/main/webapp/js/main.js]: The dialog appears when you try to delete a project, a component, or a release, for example. Search the codebase for calls of `deleteConfirmed()` to find more usages.

Comment (maximilianhuber) [inline frontend/sw360-portlet/src/main/webapp/js/main.js]: You are totally right:

old version:
withcancel

new version:
withoutcancel

I will fix this.

**Comment by mcjaeger** [comment]
it seems like the attachments JS is not working properly (tested with Safari 10.0.2)

screen shot 2017-05-03 at 15 38 15

screen shot 2017-05-03 at 15 37 27

**Comment by maximilianhuber** [comment]
Wen trying to upload I see the following in the log:
screenshot
Either we make a false request or the backend answers with 404 to a correct request.

File: frontend/sw360-portlet/src/main/webapp/js/main.js (hunk 1 above)
Comment (maximilianhuber) [inline frontend/sw360-portlet/src/main/webapp/js/main.js]: This problem is now fixed with the lines [247ff](https://github.com/sw360/sw360portal/pull/369/commits/190402e6e0ac5dc526ebfc541ee4800004bfdb85#diff-b2d4b3e262a033e0e8c3a1a5e818d4ccR247)

**Comment by maximilianhuber** [comment]
The file `resumable.js` is changed as follow:
```diff
49d48
<       parameterNamespace:'',
100c99
< 
---
>     
122,123c121,122
< 
< 
---
>     
>     
199c198
<     var preventDefault = function(e) {
---
>     var onDragOver = function(e) {
207c206
<      *            That capability is only currently available in Chrome, but if it isn't available it will just pass the items along to
---
>      *            That capability is only currently available in Chrome, but if it isn't available it will just pass the items along to 
280,306c279,291
<             var newEntries = [];
<             //wrap the callback in another function so we can store the path in a closure
<             var readDir = function(path){
<                 reader.readEntries(
<                     //success callback: read entries out of the directory
<                     function(entries){
<                         if (entries.length>0){
<                             //add these results to the array of all the new stuff
<                             for (var i=0; i<entries.length; i++) { newEntries.push(entries[i]); }
<                             //call this function again as all the results may not have been sent yet
<                             readDir(entry.fullPath);
<                         }
<                         else {
<                             //we have now gotten all the results in newEntries so let's process them recursively
<                             loadFiles(newEntries, event, queue, path);
<                             //this was a directory rather than a file so decrement the expected file count
<                             updateQueueTotal(-1, queue);
<                         }
<                     },
<                     //error callback, most often hit if there is a directory with nothing inside it
<                     function(err) {
<                         //this was a directory rather than a file so decrement the expected file count
<                         updateQueueTotal(-1, queue);
<                         console.warn(err);
<                     }
<                 );
<             };
---
>           //wrap the callback in another function so we can store the path in a closure
>           var readDir = function(path){
>             return function(entries){
>               //process each thing in this directory recursively
>               loadFiles(entries, event, queue, path);
>               //this was a directory rather than a file so decrement the expected file count
>               updateQueueTotal(-1, queue);
>             }
>           };
> 
>           reader.readEntries(readDir(entry.fullPath), function(err) {
>             console.warn(err);
>           });
319c304
< 
---
>       
348c333
<         // if single-file upload, file is already added, and trying to add 1 new file, simply replace the already-added file
---
>         // if single-file upload, file is already added, and trying to add 1 new file, simply replace the already-added file 
358,372c343,349
<         var fileName = file.name;
<         if(o.fileType.length > 0){
< 			var fileTypeFound = false;
< 			for(var index in o.fileType){
< 				var extension = '.' + o.fileType[index];
< 				if(fileName.indexOf(extension, fileName.length - extension.length) !== -1){
< 					fileTypeFound = true;
< 					break;
< 				}
< 			}
< 			if (!fileTypeFound) {
< 			  o.fileTypeErrorCallback(file, errorCount++);
< 			  return false;
< 			}
< 		}
---
>         var fileName = file.name.split('.');
>         var fileType = fileName[fileName.length-1].toLowerCase();
>         
>         if (o.fileType.length > 0 && !$h.contains(o.fileType, fileType)) {
>           o.fileTypeErrorCallback(file, errorCount++);
>           return false;
>         }
383,386c360,363
<         function addFile(uniqueIdentifier){
<           if (!$.getFromUniqueIdentifier(uniqueIdentifier)) {(function(){
<             file.uniqueIdentifier = uniqueIdentifier;
<             var f = new ResumableFile($, file, uniqueIdentifier);
---
>         // directories have size == 0
>         if (!$.getFromUniqueIdentifier($h.generateUniqueIdentifier(file))) {(function(){
>           var f = new ResumableFile($, file);
>           window.setTimeout(function(){
390,408c367,369
<             window.setTimeout(function(){
<               $.fire('fileAdded', f, event)
<             },0);
<           })()};
<         }
<         // directories have size == 0
<         var uniqueIdentifier = $h.generateUniqueIdentifier(file)
<         if(uniqueIdentifier && typeof uniqueIdentifier.done === 'function' && typeof uniqueIdentifier.fail === 'function'){
<           uniqueIdentifier
<           .done(function(uniqueIdentifier){
<               addFile(uniqueIdentifier);
<           })
<           .fail(function(){
<               addFile();
<           });
<         }else{
<           addFile(uniqueIdentifier);
<         }
< 
---
>             $.fire('fileAdded', f, event)
>           },0);
>         })()};
416c377
<     function ResumableFile(resumableObj, file, uniqueIdentifier){
---
>     function ResumableFile(resumableObj, file){
426c387
<       $.uniqueIdentifier = uniqueIdentifier;
---
>       $.uniqueIdentifier = $h.generateUniqueIdentifier(file);
429c390
<       var _error = uniqueIdentifier !== undefined;
---
>       var _error = false;
534c495
<       };
---
>       };    
564d524
< 
613,614c573
<         var parameterNamespace = $.getOpt('parameterNamespace');
<         var customQuery = $.getOpt('query');
---
>         var customQuery = $.getOpt('query'); 
617c576
<           params.push([encodeURIComponent(parameterNamespace+k), encodeURIComponent(v)].join('='));
---
>           params.push([encodeURIComponent(k), encodeURIComponent(v)].join('='));
620,628c579,587
<         params.push([parameterNamespace+'resumableChunkNumber', encodeURIComponent($.offset+1)].join('='));
<         params.push([parameterNamespace+'resumableChunkSize', encodeURIComponent($.getOpt('chunkSize'))].join('='));
<         params.push([parameterNamespace+'resumableCurrentChunkSize', encodeURIComponent($.endByte - $.startByte)].join('='));
<         params.push([parameterNamespace+'resumableTotalSize', encodeURIComponent($.fileObjSize)].join('='));
<         params.push([parameterNamespace+'resumableType', encodeURIComponent($.fileObjType)].join('='));
<         params.push([parameterNamespace+'resumableIdentifier', encodeURIComponent($.fileObj.uniqueIdentifier)].join('='));
<         params.push([parameterNamespace+'resumableFilename', encodeURIComponent($.fileObj.fileName)].join('='));
<         params.push([parameterNamespace+'resumableRelativePath', encodeURIComponent($.fileObj.relativePath)].join('='));
<         params.push([parameterNamespace+'resumableTotalChunks', encodeURIComponent($.fileObj.chunks.length)].join('='));
---
>         params.push(['resumableChunkNumber', encodeURIComponent($.offset+1)].join('='));
>         params.push(['resumableChunkSize', encodeURIComponent($.getOpt('chunkSize'))].join('='));
>         params.push(['resumableCurrentChunkSize', encodeURIComponent($.endByte - $.startByte)].join('='));
>         params.push(['resumableTotalSize', encodeURIComponent($.fileObjSize)].join('='));
>         params.push(['resumableType', encodeURIComponent($.fileObjType)].join('='));
>         params.push(['resumableIdentifier', encodeURIComponent($.fileObj.uniqueIdentifier)].join('='));
>         params.push(['resumableFilename', encodeURIComponent($.fileObj.fileName)].join('='));
>         params.push(['resumableRelativePath', encodeURIComponent($.fileObj.relativePath)].join('='));
>         params.push(['resumableTotalChunks', encodeURIComponent($.fileObj.chunks.length)].join('='));
650c609
<           case 0: $.preprocessState = 1; preprocess($); return;
---
>           case 0: preprocess($); $.preprocessState = 1; return;
685c644
<             var retryInterval = $.getOpt('chunkRetryInterval');
---
>             var retryInterval = $.getOpt('chunkRetryInterval');          
718c677
<         bytes  = $.fileObj.file[func]($.startByte,$.endByte),
---
>         bytes  = $.fileObj.file[func]($.startByte,$.endByte), 
721,722c680
< 
<         var parameterNamespace = $.getOpt('parameterNamespace');
---
>         
728c686
<             params.push([encodeURIComponent(parameterNamespace+k), encodeURIComponent(v)].join('='));
---
>             params.push([encodeURIComponent(k), encodeURIComponent(v)].join('='));
735c693
<             data.append(parameterNamespace+k,v);
---
>             data.append(k,v);
737c695
<           data.append(parameterNamespace+$.getOpt('fileParameterName'), bytes);
---
>           data.append($.getOpt('fileParameterName'), bytes);
739c697
< 
---
>         
898,899c856
<         domNode.addEventListener('dragover', preventDefault, false);
<         domNode.addEventListener('dragenter', preventDefault, false);
---
>         domNode.addEventListener('dragover', onDragOver, false);
907,908c864
<         domNode.removeEventListener('dragover', preventDefault);
<         domNode.removeEventListener('dragenter', preventDefault);
---
>         domNode.removeEventListener('dragover', onDragOver);
```

**Comment by maximilianhuber** [comment]
@alexbrdn @mcjaeger this PR is now updated and uses the new resumable.js in version 1.0.3. I hope it works now

**Comment by alexbrdn** [comment]
Uploading attachments works now.
The deletion confirmation dialog has both buttons now, but they both look grey and lifeless. Perhaps you could restore their lively colors as well - that'd be great.

**Comment by bs-jokri** [comment]
@alexbrdn @mcjaeger Would it be ok for you to move the update of the deletion confirmation dialog colors to a new ticket. Because this work will require some changes to the CSS files and is not directly related to the update of the dependencies.

**Comment by alexbrdn** [comment]
@bs-jokri , fine with me

**Comment by mcjaeger** [comment]
I am mergint his according to the telco we have had today. Hope this is fine. (-:

**Pull request merged**